quinn-proto/src/connection/paths.rs

```diff
@@ -156,6 +171,37 @@ impl PathData {
         self.in_flight.remove(packet);
         true
     }
+
+    /// Updates the last observed address report received on this path.
+    ///
+    /// If the address was updated, it's returned to be informed to the application.
+    #[must_use = "updated observed address must be reported to the application"]
+    pub(super) fn update_observed_addr_report(
+        &mut self,
+        observed: ObservedAddr,
+    ) -> Option<SocketAddr> {
+        match self.last_observed_addr_report.as_mut() {
```

Comment:
I think this could be shorted a lot. `ObservedAddr` is copy. Maybe something like this?

```rs
let prev = self.last_observed_addr_report;
if prev.map(|prev| prev.seq_no < observed.seq_no).unwrap_or(true) {
    self.last_observed_addr_report = Some(observed);
}
if Some(observed.socket_addr()) != prev.map(|prev| prev.socket_addr()) {
    Some(observed.socket_addr())
} else {
    None
}
```
It could be golfed more but something like the above perhaps is at a nice spot for readability.